sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -97,6 +97,38 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(actual(0) == cases)
   }
 
+  test("SPARK-18091: split large if expressions into blocks due to JVM code size limit") {
+    val row = create_row("afafFAFFsqcategory2dadDADcategory8sasasadscategory24", 0)
```

Comment:
An example looks like the following test case. I've tested it. It failed in current master branch due to the 64kb issue.

    test("split large if expression") {
      val conditionExpr = (1 to 1000).map { _ =>
        And(GreaterThan(new Rand(), Literal(0.5)), LessThan(new Rand(), Literal(0.8)))
      }.reduce(And)

      val expressions = Seq(If(conditionExpr, Literal(true), Literal(false)))
      val plan = GenerateUnsafeProjection.generate(expressions, true)
      // Rand needs initialization.
      plan.initialize(1)
      val actual = plan(new GenericInternalRow(10)).toSeq(expressions.map(_.dataType))
      val expected = Seq(false)
      if (!checkResult(actual, expected)) {
        fail(s"Incorrect Evaluation: expressions: $expressions, actual: $actual, expected: $expected")
      }
    }